baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1563 @@
+							Type:    "InternalIP",
+						},
+					},
+				},
+			},
+			BMMachine: capbm.BareMetalMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mybmmachine",
+					Namespace: "myns",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "BMMachine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Status: capbm.BareMetalMachineStatus{
+					MachineStatus: capi.MachineStatus{
+						Addresses: []capi.MachineAddress{
+							capi.MachineAddress{
+								Address: "192.168.1.1",
+								Type:    "InternalIP",
+							},
+							capi.MachineAddress{
+								Address: "172.0.20.2",
+								Type:    "InternalIP",
+							},
+						},
+					},
+					Ready: true,
+				},
+			},
+			ExpectedMachine: capi.Machine{
+				Status: capi.MachineStatus{
+					Addresses: []capi.MachineAddress{
+						capi.MachineAddress{
+							Address: "192.168.1.1",
+							Type:    "InternalIP",
+						},
+						capi.MachineAddress{
+							Address: "172.0.20.2",
+							Type:    "InternalIP",
+						},
+					},
+				},
+			},
+		},
+		"Machine status unchanged, status set empty": {
+			Host: &bmh.BareMetalHost{},
+			Machine: &capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mymachine",
+					Namespace: "myns",
+				},
+				Status: capi.MachineStatus{},
+			},
+			BMMachine: capbm.BareMetalMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mybmmachine",
+					Namespace: "myns",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "BMMachine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+				Status: capbm.BareMetalMachineStatus{
+					MachineStatus: capi.MachineStatus{},
+					Ready:         true,
+				},
+			},
+			ExpectedMachine: capi.Machine{
+				Status: capi.MachineStatus{},
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Logf("## TC-%s ##", name)
+		c := fakeclient.NewFakeClientWithScheme(scheme, &tc.BMMachine)
+
+		actuator, err := newMachineManager(c, nil, nil, tc.Machine, &tc.BMMachine)
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = actuator.updateMachineStatus(context.TODO(), tc.Host)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+		}
+
+		key := client.ObjectKey{
+			Name:      tc.BMMachine.Name,
+			Namespace: tc.BMMachine.Namespace,
+		}
+
+		bmmachine := capbm.BareMetalMachine{}
+		err = c.Get(context.TODO(), key, &bmmachine)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if tc.BMMachine.Status.MachineStatus.Addresses != nil {
+			for i, address := range tc.ExpectedMachine.Status.Addresses {
+				if address != bmmachine.Status.MachineStatus.Addresses[i] {
+					t.Errorf("expected Address %v, found %v", address, bmmachine.Status.MachineStatus.Addresses[i])
+				}
+			}
+		}
+	}
+}
+
+func TestNodeAddresses(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := capi.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	nic1 := bmh.NIC{
+		IP: "192.168.1.1",
+	}
+
+	nic2 := bmh.NIC{
+		IP: "172.0.20.2",
+	}
+
```

Comment:
Address definitions are not modified to use BareMetalMachine since the BareMetalMachineStatus contains the same capi.MachineAddress information which is presented below: